Documentation/ApiOverview/DependencyInjection/Index.rst

```diff
@@ -476,13 +476,17 @@ autoconfigure
     available from the Symfony container and marked as shared
     (:yaml:`shared: true`).
 
-Model exclusion
+.. _dependency-injection-exclusion:
+
+Classpath Exclusion
     The path exclusion :yaml:`exclude: '../Classes/Domain/Model/*'` excludes
     your models from the dependency injection container, which means you cannot inject them
     nor inject dependencies into them. Models are not services and therefore
     should not require dependency injection. Also, these objects are created by
     the Extbase persistence layer, which does not support the DI container.
-
+    There are other possible error cases with classes called from outside of an extension
+    or too early during the boot process. It can be necessary to exclude several classes at once with
+    :yaml:`exclude: '../Classes/{Model,Middleware,UserFunc,Api}/*'`  
```

Comment:
Middleware and Api might be not good cases, as they are mostly services and use DI. UserFunc also, you can use DI there (and should, must just be set public). I suggest to use other examples, like `{Model, Entity}`.

Reply:
Middleware classes have cost me hours of looking for the reason of the error with dependency injection.  An Api class method should have all needed parameters passed. This also cost me many hours. I have no other idea how to solve those issues when DI throws an exception.

Reply:
Don't know what you are doing: I use many PSR-15 middlewares with DI and also API services with DI. Those should not be excluded, otherwise they cannot be used in DI context. The exclude example is bad practise for users which will raise other problems.

Reply:
This is an important distinction by Chris. Indeed it should be generally possible to use CI in Middlewares and APIs, especially when typehinting with Services is involved. So we cannot generally give the impression that these should be excluded, because when people copy and paste it without knowing why they exclude middleware etc, that can lead to much other confusion.

Maybe we can find a general phrase which 

* mentions that if your Classes have contructors not based on injectable classes, you may want to exclude them from DI
* mention a syntax example on how multiple classes can be excluded
* give at least one more precise example other than a Domain Model where exclusion may be helpful ("external API classes unrelated to Dependency Injection" or something like that)

I'm supporting Chris, that we should not conceal/hide away issues by promoting exclusions, whern DI ist just not compatible with specfic implementation or pattern decisions.

And I'm with Franz that we need better pointers for people where DI "breaks" and those people either a.) want to bypasse the system because their API design just is not fully compatible with DI or b.) want to really understand what their underlying issue is, and WHY it "breaks". That's hard to formulate, because we actually want to promote "good DI usage" and not "bypassing bad DI usage".

Reply:
I have now changed the folders into 4 neutral names. So no user will be lead in any direction.